src/yaml-language-support/yaml-support.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+
+import { parse, findNodeAtPosition, util as yamlUtil } from 'node-yaml-parser';
+
+export interface YamlCache {
+    // the document model list represents the yaml text
+    yamlDocs: any[];
+
+    // lineLens contains the converted line length of each lines, it is used for converting from
+    // vscode position to the inner position in yaml element model.
+    lineLens: number[];
+
+    // the version of the document to avoid duplicate work on the same text
```

Comment:
"the document" implies a YamlCache represents a document, not a cache.  Should we call it something like YamlDocument or YamlCachedDocument or YamlParsedDocument or something like that?

Reply:
the yamlDocs represents a parsed results of the yaml file, I keep it in a cache to avoid duplicate parse works since a number of places need the parsed results of the yaml file, eg: hover, complete, schema provider.